efi/preinstall/checks_context.go

```diff
@@ -163,7 +168,15 @@ type RunChecksContext struct {
 	// - unavailable (a value of false).
 	// - available (a value of true).
 	availableActions map[Action]bool
-	expectedActions  []Action
+
+	// expectedActions contains a slice of actions that are expected on a subsequent call
+	// to Run. Trying to execute an action that is not in here will result in an error
+	// being returned.
+	expectedActions []Action
+
+	// proceedFlags indicates the CheckFlags that will be enabled if Run is called
+	// with ActionProceed.
+	proceedFlags CheckFlags
```

Comment:
is plan to remember to always pass ActionProceed or will there be a way to extract/record these? in theory asking to proceed at a later point might be different than remember the flags for the errors the user reviewed

Reply:
I intend to add this information to a new field in `CheckResult` in a follow-up PR, so that it will be possible to remember which errors the user reviewed and accepted in the past.